src/LinuxTracing/SwitchesStatesNamesVisitorTest.cpp

```diff
@@ -185,13 +226,26 @@ ::testing::Matcher<SchedulingSlice> SchedulingSliceEq(const SchedulingSlice& exp
                           expected.out_timestamp_ns()));
 }
 
-ThreadStateSlice MakeThreadStateSlice(uint32_t tid, ThreadStateSlice::ThreadState thread_state,
-                                      uint64_t duration_ns, uint64_t end_timestamp_ns) {
+ThreadStateSlice MakeThreadStateSlice(
+    uint32_t tid, ThreadStateSlice::ThreadState thread_state, uint64_t duration_ns,
+    uint64_t end_timestamp_ns,
+    ThreadStateSlice::WakeupReason wakeup_reason = ThreadStateSlice::kNotApplicable,
+    uint32_t wakeup_tid = 0, uint32_t wakeup_pid = 0,
+    bool has_switch_out_or_wakeup_callstack = false) {
   ThreadStateSlice thread_state_slice;
   thread_state_slice.set_tid(tid);
   thread_state_slice.set_thread_state(thread_state);
   thread_state_slice.set_duration_ns(duration_ns);
   thread_state_slice.set_end_timestamp_ns(end_timestamp_ns);
+  thread_state_slice.set_wakeup_reason(wakeup_reason);
+  thread_state_slice.set_wakeup_tid(wakeup_tid);
+  thread_state_slice.set_wakeup_pid(wakeup_pid);
+  if (has_switch_out_or_wakeup_callstack) {
+    thread_state_slice.set_switch_out_or_wakeup_callstack_status(
+        ThreadStateSlice::kWaitingForCallstack);
+  } else {
+    thread_state_slice.set_switch_out_or_wakeup_callstack_status(ThreadStateSlice::kNoCallstack);
+  }
```

Comment:
I think we should also set the callstack id (to 0 I guess) if we compare the values in `ThreadStateSliceEq`.

Reply:
Done.